source/common/stats/tag_producer_impl.cc

```diff
@@ -73,6 +66,13 @@ int TagProducerImpl::addExtractorsMatching(absl::string_view name) {
 }
 
 void TagProducerImpl::addExtractor(TagExtractorPtr extractor) {
+  auto insertion = extractor_map_.insert(std::make_pair(extractor->name(), std::ref(*extractor)));
+  if (!insertion.second) {
+    extractor->setOtherExtractorWithSameNameExists(true);
```

Comment:
I did a quick scale test and I think I may drop this hack -- capturing the `set<string_view>` unconditionally during tag-extraction only costs about 2% when initializing 100k clusters, and that may be measurement error.

Reply:
Actually I'm leaning toward leaving it in. It's not *that* bad a hack to save a verified 2%. I ran a bunch of "envoy-static --mode init_only on a 10k cluster static config file and it saves an average of 2%.

If you think that's not worth it I can remove it and the population of TagExtractor::otherExtractorWithSameNameExists().